exporter/datadogexporter/metrics_exporter_test.go

```diff
@@ -982,6 +986,102 @@ func Test_metricsExporter_PushMetricsData_Zorkian(t *testing.T) {
 	}
 }
 
+func TestNewExporterWithProxy(t *testing.T) {
+	server := testutil.DatadogServerMock()
+	defer server.Close()
+
+	var proxyRequests []*http.Request
+
+	proxyServer := httptest.NewServer(&httputil.ReverseProxy{
+		Director: func(req *http.Request) {
+			proxyRequests = append(proxyRequests, req)
+
+			req.Header.Set("X-Proxy-Test", "test-proxy-123")
+			req.URL.Scheme = "http"
+			req.URL.Host = server.Listener.Addr().String()
+		},
+	})
+	defer proxyServer.Close()
+
+	cfg := &datadogconfig.Config{
+		API: datadogconfig.APIConfig{
+			Key: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+		},
+		Metrics: datadogconfig.MetricsConfig{
+			TCPAddrConfig: confignet.TCPAddrConfig{
+				Endpoint: proxyServer.URL,
+			},
+			DeltaTTL: 3600,
+			HistConfig: datadogconfig.HistogramConfig{
+				Mode:             datadogconfig.HistogramModeDistributions,
+				SendAggregations: false,
+			},
+			SumConfig: datadogconfig.SumConfig{
+				CumulativeMonotonicMode: datadogconfig.CumulativeMonotonicSumModeToDelta,
+			},
+		},
+		HostMetadata: datadogconfig.HostMetadataConfig{
+			Enabled:        true,
+			ReporterPeriod: 30 * time.Minute,
+		},
+		HostnameDetectionTimeout: 50 * time.Millisecond,
+
+		ClientConfig: confighttp.ClientConfig{
+			ProxyURL: proxyServer.URL,
+		},
+	}
+
+	params := exportertest.NewNopSettings(metadata.Type)
+	f := NewFactory()
+
+	// The client should have been created correctly
+	exp, err := f.CreateMetrics(context.Background(), params, cfg)
+	require.NoError(t, err)
+	assert.NotNil(t, exp)
+
+	// Create & send test metrics (no metadata)
+	testMetrics := pmetric.NewMetrics()
+	testutil.TestMetrics.CopyTo(testMetrics)
+	err = exp.ConsumeMetrics(context.Background(), testMetrics)
+	require.NoError(t, err)
+	assert.Empty(t, server.MetadataChan)
+
+	// Send another with metadata
+	testMetrics = pmetric.NewMetrics()
+	testutil.TestMetrics.CopyTo(testMetrics)
+	err = exp.ConsumeMetrics(context.Background(), testMetrics)
+	require.NoError(t, err)
```

Comment:
You can call `exp.Shutdown` then `proxyServer.Close()`, `server.Close()` after this line